python/simulation.py

```diff
@@ -98,6 +98,32 @@ def py_v3_to_vec(dims, iterable, is_cylindrical=False):
     else:
         raise ValueError("Invalid dimensions in Volume: {}".format(dims))
 
+class DiffractedPlanewave(object):
```

Comment:
Note: the current plan is to optionally pass this for the `band_num` parameter to eigenmode sources etcetera.